### PR TITLE
Make transformers configurable

### PR DESCRIPTION
**Pull request by bhirsz**
Implements #4 

We're still in draft phase so all of our code is subject to big changes - but please comment if it's going in good/bad directotion. Every input is welcome!

Added `@transform` decorator - only transformers with this decorator will be automatically loaded (it's step in direction of explicit loading). 
When none transformer is set with `--transform` option robotidy will load all transformer with `@transform` decorator set. Otherwise it will filter out by --transform names.

Added `@configurable` decorator - it behaves similarly as Python property (with the difference that we should return value from our decorated function. Allow to change property value through cli configuration. For example this class:
```
@transformer
class SomeClass:
    def __init__(self):
        self.my_param = 5

    @configurable
    def my_param(self, value):
         parsed = int(value) + 5
         return parsed
```
invoked with:
```
robotidy --transform SomeClass:my_param:30
```

Will initialize SomeClass with my_param = 5, then it will load the value from configuration (int(30) + 5)  = 35. When accessed later my_param will return 35:
```
def transform(self, model):
     print(self.some_param)
> 35
```

### CODE REVIEW

**Comment by bhirsz** [comment]
Another step will be creating testing for robotidy - only then I will be able to extend proposed features (and also support any edge case, exceptions).

File: robotidy/decorators.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class configurable:  # noqa
+    def __init__(self, fun):
+        self.fun = fun
+
+    def __set_name__(self, owner, name):
+        if not hasattr(owner, 'configurables'):
+            owner.configurables = set()
+        owner.configurables.add(name)
```
Comment (bhirsz) [inline robotidy/decorators.py:8]: Here we're 'registering' configurable name in transformer class - so it will raise exception if someone will try to configure not existing param:
```
robotidy  --transform MyTransform:i_dont_exist:5
ValueError("Invalid configurable name: ':i_dont_exist' for transformer: 'MyTransform'")
```
